### PR TITLE
[MM-61122] Add diagnostics data to the Support Packet

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
When a Support Packet gets generated, the plugin now includes diagnostics data via a `diagnostics.yaml` file into the Packet.

Example content of `com.mattermost.calls/diagnostics.yaml`:

```yaml
version: 1.2.0+b5a4211
total_calls: 1
total_active_calls: 1
total_active_sessions: 2
avg_duration: 51
avg_participants: 2
```

Requires https://github.com/mattermost/mattermost/pull/28833 on the server side.

See https://mattermost.atlassian.net/wiki/spaces/CLOUD/pages/3083108354/Diagnostics+data+from+Core+Plugins+to+the+Support+Packet for more context.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-61122

### CODE REVIEW

File: server/support_packet.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	// The average calls duration in seconds.
+	AvgDuration int64 `yaml:"avg_duration"`
+	// The average peak number of participants in calls.
+	AvgParticipants int64 `yaml:"avg_participants"`
+}
+
+func (p *Plugin) GenerateSupportData(_ *plugin.Context) ([]*model.FileData, error) {
+	stats, err := p.store.GetCallsStats()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to calls stats")
```
Comment (streamer45) [inline server/support_packet.go:33]:
```suggestion
		return nil, errors.Wrap(err, "Failed to get calls stats")
```

File: server/support_packet.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+	// The total number calls.
+	TotalCalls int64 `yaml:"total_calls"`
+	// The total number of active calls.
+	TotalActiveCalls int64 `yaml:"total_active_calls"`
+	// The total number of active sessions.
+	TotalActiveSessions int64 `yaml:"total_active_sessions"`
+	// The average calls duration in seconds.
+	AvgDuration int64 `yaml:"avg_duration"`
+	// The average peak number of participants in calls.
+	AvgParticipants int64 `yaml:"avg_participants"`
```
Comment (streamer45) [inline server/support_packet.go:27]: Not overly in love with the repetition here. Wondering if we could embed `CallsStats` or perhaps create a shared [public](https://github.com/mattermost/mattermost-plugin-calls/tree/main/server/public) subtype.

**Review by streamer45 — COMMENTED** [review]
Thanks @hanzei . A couple of comments below.

File: server/support_packet.go (hunk 2 above)
Comment (hanzei) [reply server/support_packet.go:27]: I'm not so sure if we should combine these two data structs. While they share common data, they don't share common uses. Is it worth avoiding a copy by coupling them?

**Review by streamer45 — APPROVED** [review]
Looks great, thanks!

**Comment by hanzei** [comment]
@wiggin77 Gentle reminder to review the PR :wink:

File: server/support_packet.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+type SupportPacket struct {
+	Version string `yaml:"version"`
+
+	*public.CallsStats
+}
+
+func (p *Plugin) GenerateSupportData(_ *plugin.Context) ([]*model.FileData, error) {
+	stats, err := p.store.GetCallsStats()
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to get calls stats")
```
Comment (wiggin77) [inline server/support_packet.go:26]: Errors should be lowercase.
```suggestion
		return nil, errors.Wrap(err, "failed to get calls stats")
```

File: server/support_packet.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	}
+
+	diagnostics := SupportPacket{
+		Version:    manifest.Version,
+		CallsStats: stats,
+	}
+	body, err := yaml.Marshal(diagnostics)
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to marshal diagnostics")
```
Comment (wiggin77) [inline server/support_packet.go:35]:
```suggestion
		return nil, errors.Wrap(err, "failed to marshal diagnostics")
```

**Review by wiggin77 — APPROVED** [review]
Approved with minor change request.

**Comment by hanzei** [comment]
Per discussion with @raghavaggarwal2308, QA testing will happen post-merge.

**Pull request merged**